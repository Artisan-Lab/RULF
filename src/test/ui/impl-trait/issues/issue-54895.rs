trait Trait<'a> {
    type Out;
    fn call(&'a self) -> Self::Out;
}

struct X(());

impl<'a> Trait<'a> for X {
    type Out = ();
    fn call(&'a self) -> Self::Out {
        ()
    }
}

fn f() -> impl for<'a> Trait<'a, Out = impl Sized + 'a> {
    //~^ ERROR higher kinded lifetime bounds on nested opaque types are not supported yet
    X(())
}

fn main() {
    let _ = f();
}
