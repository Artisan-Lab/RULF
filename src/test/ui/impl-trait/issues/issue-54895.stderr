error: higher kinded lifetime bounds on nested opaque types are not supported yet
  --> $DIR/issue-54895.rs:15:53
   |
LL | fn f() -> impl for<'a> Trait<'a, Out = impl Sized + 'a> {
   |                                                     ^^
   |
note: lifetime declared here
  --> $DIR/issue-54895.rs:15:20
   |
LL | fn f() -> impl for<'a> Trait<'a, Out = impl Sized + 'a> {
   |                    ^^

error: aborting due to previous error

